extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/write/HiveWrite.scala

```diff
@@ -0,0 +1,243 @@
+      fileSinkConf.setCompressCodec(hadoopConf
+        .get("mapreduce.output.fileoutputformat.compress.codec"))
+      fileSinkConf.setCompressType(hadoopConf
+        .get("mapreduce.output.fileoutputformat.compress.type"))
+    } else {
+      // Set compression by priority
+      HiveOptions.getHiveWriteCompression(fileSinkConf.getTableInfo, sparkSession.sessionState.conf)
+        .foreach { case (compression, codec) => hadoopConf.set(compression, codec) }
+    }
+  }
+
+  private def validateExtractPartitionCols(
+      fileSinkConf: FileSinkDesc,
+      hadoopConf: Configuration): Seq[String] = {
+    val partitionSpec = getPartitionSpec(partition)
+    val numDynamicPartitions = partitionSpec.values.count(_.equals(""))
+    val numStaticPartitions = partitionSpec.size - numDynamicPartitions
+
+    // All partition column names in the format of "<column name 1>/<column name 2>/..."
+    val partitionColumns = fileSinkConf.getTableInfo.getProperties.getProperty("partition_columns")
+    val partitionColumnNames = Option(partitionColumns).map(_.split("/")).getOrElse(Array.empty)
+
+    // Validate partition spec if there exist any dynamic partitions
+    if (numDynamicPartitions > 0) {
+      // Report error if dynamic partitioning is not enabled
+      if (!hadoopConf.get("hive.exec.dynamic.partition", "true").toBoolean) {
+        throw KyuubiHiveConnectorException(ErrorMsg.DYNAMIC_PARTITION_DISABLED.getMsg)
+      }
+
+      // Report error if dynamic partition strict mode is on but no static partition is found
+      if (numStaticPartitions == 0 &&
+        hadoopConf.get("hive.exec.dynamic.partition.mode", "strict").equalsIgnoreCase("strict")) {
+        throw KyuubiHiveConnectorException(ErrorMsg.DYNAMIC_PARTITION_STRICT_MODE.getMsg)
+      }
+
+      // Report error if any static partition appears after a dynamic partition
+      val isDynamic = partitionColumnNames.map(partitionSpec(_).equals(""))
+      if (isDynamic.init.zip(isDynamic.tail).contains((true, false))) {
+        throw KyuubiHiveConnectorException(ErrorMsg.PARTITION_DYN_STA_ORDER.getMsg)
+      }
+      validateBucketSpec()
+    }
+    partitionColumnNames
+  }
+
+  private def validateBucketSpec(): Unit = {
+    table.bucketSpec match {
+      case Some(_) =>
+        // Writes to bucketed hive tables are allowed only if user does not care about maintaining
+        // table's bucketing i.e. both "hive.enforce.bucketing" and "hive.enforce.sorting" are
+        // set to false
+        val enforceBucketingConfig = "hive.enforce.bucketing"
+        val enforceSortingConfig = "hive.enforce.sorting"
+
+        val message = s"Output Hive table ${table.identifier} is bucketed but Spark " +
+          "currently does NOT populate bucketed output which is compatible with Hive."
+
+        if (hadoopConf.get(enforceBucketingConfig, "true").toBoolean ||
+          hadoopConf.get(enforceSortingConfig, "true").toBoolean) {
+          throw KyuubiHiveConnectorException(message)
```

Comment:
Spark removed this limitation after SPARK-32709

Reply:
do we need some actions to make it work? or just need to remove the check?

Reply:
Hi @Yikf, have you checked that?

Reply:
Is the UT included in this PR? I can not find it by searching the keyword "bucket"